remodel/models.py

```diff
@@ -30,24 +31,22 @@ def __new__(cls, name, bases, dct):
             dict(rel_attrs, model=name))
         object_handler_cls = dct.setdefault('object_handler', ObjectHandler)
         
-        new_class = super_new(cls, name, bases, dct)
+        new_class = super_new(mcs, name, bases, dct)
         model_registry.register(name, new_class)
         setattr(new_class, 'objects', object_handler_cls(new_class))
         return new_class
 
     # Proxies undefined attributes to Model.objects; useful for building
     # ReQL queries directly on the Model (e.g.: User.order_by('name').run())
-    def __getattr__(cls, name):
-        return getattr(cls.objects, name)
-
+    def __getattr__(self, name):
+        return getattr(self.objects, name)
 
+@add_metaclass(ModelBase)
 class Model(object):
-    __metaclass__ = ModelBase
-
     def __init__(self, **kwargs):
         self.fields = self._field_handler_cls()
 
-        for key, value in kwargs.items():
+        for key, value in list(kwargs.items()):
```

Comment:
We can keep the old syntax, both for clarity and speed. There is no need in the loop for the dict view to be transformed to a list.